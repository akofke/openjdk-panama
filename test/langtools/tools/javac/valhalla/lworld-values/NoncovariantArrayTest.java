/*
 * @test /nodynamiccopyright/
 * @bug 8215507
 * @summary javac should forbid conversion from value array to Object[]
 * @compile/fail/ref=NoncovariantArrayTest.out -XDrawDiagnostics -XDdev NoncovariantArrayTest.java
 */

public class NoncovariantArrayTest { 
    static final value class V { 
        public final int v1; 
        private V () {v1 = 0;} 
    } 

    public static void main(String args[]) { 
        int [] ia = new int[1]; 
        Object oa[] = (Object[])ia; 
        oa = ia;

        V [] va = new V[1]; 
        Object oa2[] = (Object[])va; 
        oa2 = va; 
        va = oa2;
        va = (V []) oa2;
    } 
}