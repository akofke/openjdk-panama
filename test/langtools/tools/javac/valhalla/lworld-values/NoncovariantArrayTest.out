NoncovariantArrayTest.java:16:33: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: int[], java.lang.Object[])
NoncovariantArrayTest.java:17:14: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: int[], java.lang.Object[])
NoncovariantArrayTest.java:20:34: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: NoncovariantArrayTest.V[], java.lang.Object[])
NoncovariantArrayTest.java:21:15: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: NoncovariantArrayTest.V[], java.lang.Object[])
NoncovariantArrayTest.java:22:14: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Object[], NoncovariantArrayTest.V[])
NoncovariantArrayTest.java:23:21: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Object[], NoncovariantArrayTest.V[])
6 errors